src/wal/src/local_storage_impl/segment.rs

```diff
@@ -383,10 +465,13 @@ pub struct SegmentManager {
     all_segments: Mutex<HashMap<u64, Arc<Mutex<Segment>>>>,
 
     /// Cache for opened segments
-    cache: Mutex<VecDeque<u64>>,
+    cache: Mutex<VecDeque<(u64, Arc<Mutex<Segment>>)>>,
```

Comment:
Why add seq here?

Reply:
The `u64` in this tuple represents the segment ID. The reason for storing it is to determine if a segment exists in the cache without needing to acquire the lock for that segment. The newly added `Arc<Mutex<Segment>>` is to remove the original `SegmentManager::get_segment` method, which is redundant and may cause deadlock.